src/repository/x509.rs

```diff
@@ -1115,30 +1183,54 @@ impl fmt::Display for RepresentationError {
 impl error::Error for RepresentationError { }
 
 
-//------------ ValidationError -----------------------------------------------
+//------------ ValidityPeriodError -------------------------------------------
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub struct ValidationError;
+/// An object is outside of its period of validity.
+#[derive(Clone, Copy, Debug)]
+pub struct ValidityPeriodError {
+    /// Is the object too new?
+    ///
+    /// It is too old otherwise.
+    too_new: bool,
+}
```

Comment:
I probably would have used an enum, but that's bike shedding I suppose.. so no objection either.